Захаров Андрей

Folleach opened the pull request:
@KonstantinDobrovolsky

Folleach committed:
Create template

Folleach committed:
Remove console dependency from command

KonstantinDobrovolsky commented on TagCloud.Tests/TagCloud.Tests.csproj:
```diff
@@ -0,0 +1,7 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Класть всё в один коммит это плохая практика - тут и создание проекта, и базовых классов, и куча разной логики. Не стоит так делать. Лучше разбивать на умеренно небольшие коммиты, объединяя изменения по, как бы сказать, небольшим, конечным идеям. Так гораздо удобнее работать с хистори коммитов, если вдруг понадобиться

KonstantinDobrovolsky commented on TagCloud/Program.cs:
```diff
@@ -0,0 +1,67 @@
+            }).SingleInstance();
+            builder.Register(ctx => CloudSettings.GetDefault()).SingleInstance();
+
+            container = builder.Build();
```
Кажется, вся эта логика билдов очень хорошо выносится в отдельный метод. Не стоит засорять так main

KonstantinDobrovolsky commented on TagCloud/Program.cs:
```diff
@@ -0,0 +1,67 @@
+            }).SingleInstance();
+            builder.Register(ctx => new ResultSettings
+            {
+                OutputPath = "created/"
```
Мелочь, но кажется удобнее перенести папку чуть выше, чтобы каждый раз не лазить за результатом вглубь bin'а)

KonstantinDobrovolsky commented on TagCloud/Controllers/ConsoleController.cs:
```diff
@@ -0,0 +1,51 @@
+
+        private ICommand FindCommand(string commandId)
+        {
+            return commands.ContainsKey(commandId) ? commands[commandId] : null;
```
Может тут лучше подошел бы TryGetValue, чтобы не бегать по словарю дважды?

KonstantinDobrovolsky commented on TagCloud/Extensions/ContainerBuilderExtensions.cs:
```diff
@@ -0,0 +1,17 @@
+    {
+        public static void RegisterCommand<T>(this ContainerBuilder builder)
+        {
+            builder.RegisterType<T>().As<ICommand>().InstancePerLifetimeScope().PropertiesAutowired(PropertyWiringOptions.AllowCircularDependencies);
```
Такие длинные строки читать ужасно неудобно. Стоит разбить на несколько строк для удобства восприятия)

KonstantinDobrovolsky commented on TagCloud/Commands/ICommand.cs:
```diff
@@ -0,0 +1,14 @@
+﻿using System;
```
Понимаю, что это просто предпроверка, но на всякий случай напомню, что в конечном решении не стоит оставлять лишние юзинги. Сейчас они у тебя много где встречаются)

Folleach committed:
Refactor entry point

Folleach committed:
Create TextSource

Folleach committed:
Add word rendering

Folleach committed:
Create source command and fix text size

Folleach committed:
Create command for manipulate with cloud settings

Folleach committed:
Fix some minor bugs

Such as:
- Lower case in Words (Minor technical specifications? It isn't)
- Migrate to TryGetValue
- Remove useless usings
- Format output in help command
- Remove comma in CloudSettingsCommand
- The large line is divided into small ones

Folleach committed:
Add ability to manipulate ignore words

Folleach committed:
Create DocxSource and change ISource

Also add ignoring empty words

Folleach committed:
Add tests for Source
